File: c/easy/divide_an_array_into_subarrays_with_minimum_cost_i.c
#include <limits.h>

int minimumCost(int* nums, int size) {
    int first = nums[0];
    int min1 = INT_MAX;
    int min2 = INT_MAX;

    for (int i = 1; i < size; i++) {
        int x = nums[i];
        if (x < min1) {
            min2 = min1;
            min1 = x;
        } else if (x < min2) {
            min2 = x;
        }
    }

    return first + min1 + min2;
}

File: rust/easy/divide_an_array_into_subarrays_with_minimum_cost_i.rs
fn minimum_cost(nums: &[i32]) -> i32 {
    let first = nums[0];
    let mut min1 = i32::MAX;
    let mut min2 = i32::MAX;

    for &x in &nums[1..] {
        if x < min1 {
            min2 = min1;
            min1 = x;
        } else if x < min2 {
            min2 = x;
        }
    }

    first + min1 + min2
}
